ui/components/canvas2d/src/components/PolygonGroup.svelte

```diff
@@ -87,43 +122,63 @@
     }
   }
 
-  const hexToRGBA = (hex: string, alpha: number) => {
-    const r = parseInt(hex.slice(1, 3), 16);
-    const g = parseInt(hex.slice(3, 5), 16);
-    const b = parseInt(hex.slice(5, 7), 16);
-    return `rgba(${r},${g},${b},${alpha})`;
-  };
+  function updateCircleRadius(id: number, i: number, radius: number) {
+    const point: Konva.Circle = stage.findOne(`#dot-${i}-${id}`);
+    point.radius(radius);
+  }
 </script>
 
 <Group config={{ id: "polygon", draggable: canEdit, visible: polygonDetails.visible }}>
-  {#if !!polygonDetails.points.length}
-    <Line
+  {#if canEdit}
+    {#if polygonDetails.id === "creating"}
+      <Path
+        config={{
+          data: polygonShape.simplifiedSvg[0],
+          stroke: "hsl(316deg 60% 29.41%)",
+          strokeWidth: 2,
+          fill: "rgb(0,128,0,0.5)",
+        }}
+      />
+    {/if}
+    <KonvaShape
       config={{
-        points: flatPolygonPoints,
+        sceneFunc: (ctx, stage) => sceneFunc(ctx, stage, polygonShape.simplifiedSvg),
         stroke: polygonDetails.status === "created" ? color : "hsl(316deg 60% 29.41%)",
-        strokeWidth: polygonDetails.status === "created" ? 1 : 3,
+        strokeWidth: 1,
         closed: isCurrentPolygonClosed,
         fill: polygonDetails.status === "created" ? hexToRGBA(color, 0.5) : "rgb(0,128,0,0.5)",
       }}
     />
-  {/if}
-  {#if canEdit}
-    {#each polygonDetails.points as point, i}
-      <Circle
-        on:click={() => handlePolygonPointsClick(i, viewId)}
-        on:dragmove={() => handlePolygonPointsDragMove(point.id)}
-        on:dragend={handlePolygonPointsDragEnd}
-        config={{
-          x: point.x,
-          y: point.y,
-          radius: 5,
-          fill: "rgb(0,128,0)",
-          stroke: "white",
-          strokeWidth: 2,
-          id: `dot-${point.id}`,
-          draggable: true,
-        }}
-      />
+    {#each polygonShape.simplifiedPoints as shape, i}
+      {#each shape as point, pi}
+        <Circle
+          on:click={() => handlePolygonPointsClick(pi, viewId)}
+          on:dragmove={() => handlePolygonPointsDragMove(point.id, i)}
+          on:dragend={handlePolygonPointsDragEnd}
+          on:mouseover={() => point.id === 0 && updateCircleRadius(point.id, i, 4)}
+          on:mouseleave={() => updateCircleRadius(point.id, i, 2)}
+          config={{
+            x: point.x,
+            y: point.y,
+            radius: 2,
```

Comment:
Note for later: radius should be zoom dependant

Reply:
yes, good idea. I add an issue.

Reply:
https://github.com/orgs/pixano/projects/3/views/1?pane=issue&itemId=50133596